sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/IssueLocationCollector.cs

```diff
@@ -30,23 +30,83 @@
+    /// </code>
+    ///
+    /// Checking the precise/exact location of an issue. Only one precise location or column location can be present
+    /// at one time. Precise location is used by adding '^^^^' comment under the location where the issue is expected.
+    /// The alternative column location pattern can be used by following the 'Noncompliant' or 'Secondary' comment
+    /// with '^X#Y' where 'X' is the expected start column and Y the length of the issue.
+    /// <code>
+    ///     private void MyMethod() // Noncompliant
+    /// //  ^^^^^^^
+    /// </code>
+    /// <code>
+    ///     private void MyMethod() // Noncompliant ^4#7
+    /// </code>
+    /// 
+    /// Multiple issues per line. To declare that multiple issues are expected, each issue must be assigned an id. All
+    /// secondary locations associated with an issue must have the same id. Note that it is not possible to have multiple
+    /// precise/column locations on a single line.
+    /// <code>
+    ///     var a = null; // Noncompliant [myId2]
+    ///     if (myCondition) // Noncompliant [myId1, myId3]
+    ///     {
+    ///       a = null; // Secondary [myId1, myId2]
+    ///     }
+    /// </code>
+    ///
+    /// Note that most of the previous patterns can be used together when needed.
+    /// <code>
+    ///     private void MyMethod() // Noncompliant@+1 ^4#7 [MyIssueId] {{Remove this unused private method}}
+    /// </code>
+    /// <code>
+    /// </summary>
     public class IssueLocationCollector
     {
-        private const string OFFSET_PATTERN = @"(?:@(?<sign>[\+|-]?)(?<offset>[0-9]+))?";
-        private const string ISSUE_IDS_PATTERN = @"\s*(\[(?<issueIds>.*)\])*";
-        private const string MESSAGE_PATTERN = @"\s*(\{\{(?<message>.*)\}\})*";
-        private const string TYPE_PATTERN = @"(?<issueType>Noncompliant|Secondary)";
-        private const string BUILD_ERROR_PATTERN = @"Error";
-        private const string PRECISE_LOCATION_PATTERN = @"\s*(?<position>\^+)\s*";
-        private const string NO_PRECISE_LOCATION_PATTERN = @"\s*(?<!\^+\s{1})";
-        private const string COMMENT_PATTERN = @"(?<comment>\/\/|\')";
-
-        internal const string ISSUE_LOCATION_PATTERN =
-            COMMENT_PATTERN + "*" + NO_PRECISE_LOCATION_PATTERN + TYPE_PATTERN + OFFSET_PATTERN + ISSUE_IDS_PATTERN + MESSAGE_PATTERN;
-        private const string PRECISE_ISSUE_LOCATION_PATTERN =
-            @"^" + COMMENT_PATTERN + PRECISE_LOCATION_PATTERN + TYPE_PATTERN + "*" + OFFSET_PATTERN + ISSUE_IDS_PATTERN + MESSAGE_PATTERN + "$";
-        internal const string BUILD_ERROR_LOCATION_PATTERN =
-            COMMENT_PATTERN + NO_PRECISE_LOCATION_PATTERN + BUILD_ERROR_PATTERN + OFFSET_PATTERN + ISSUE_IDS_PATTERN;
+        private const string CommentPattern = "(?<comment>//|')";
+        private const string PrecisePositionPattern = @"\s*(?<position>\^+)(\s+(?<invalid>\^+))*\s*";
```

Comment:
The last \s* will be useless, everyone should be responsible for it's own leading white space.
```suggestion
        private const string PrecisePositionPattern = @"\s*(?<position>\^+)(\s+(?<invalid>\^+))*";
```